src/language-html/printer-html.js

```diff
@@ -244,10 +244,11 @@ function genericPrint(path, options, print) {
                           node.isIndentationSensitive)) &&
                       new RegExp(
                         `\\n\\s{${options.tabWidth *
-                          countParents(
+                          (countParents(
                             path,
                             n => n.parent && n.parent.type !== "root"
-                          )}}$`
+                          ) +
+                            (options.parentParser ? 2 : 0))}}$`
```

Comment:
We need to figure out a way to access the indentation information from the parent context since the parent context can be different, for example:

**Prettier pr-5668**
[Playground link](https://deploy-preview-5668--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEALGBbANgAwDpQAEhAPAA4BOcAfAccQB52kD0lNB+UBAlgGaEAFDAoBXOAEpCwZumxd6pdrSKLCTVcRJsqK4lwC+BXgOFjJ05vyEjxUmZsJzczReV2u1Gta2WeFhEZQBiAANCAQZDA80ADOyKAAhhQUEADuAArJCPEoiQBuEDwAJmEgAEYUiWAA1nAwAMpk1TxQAObItnDhzgDqqDzwsc1gcA05gzz5gwCeyOCx8eGtsXAUMBlVbRiJyHyJWKvhAFaxDABCVbX1DYkYcAAyrXB7B0cgpwwNrW1YcACKogg8Feh26IGaFFWFHmMBmZDgsTAFB4UTKlFaMF6JRgqGQAA4AAzhSgQVa9KpkebsaH5F7hKgAR1EPCom0S212SH2YPCqwwPE65j5Pz+gOBL25b3BMES5WxxVxyAATOERIkeFgfgBhCAYHbzKDQekgUSrAAqctyPNWBgMQA)
```sh
--parser typescript
```

**Input:**
```tsx
html`
  <pre>
    x
  </pre>
`

if (true) {
  html`
    <pre>
      x
    </pre>
  `
}

if (true) {
  if (true) {
    html`
      <pre>
        x
      </pre>
    `
  }
}
```

**Output:**
```tsx
html`
  <pre>
    x
  </pre
  >
`;

if (true) {
  html`
    <pre>
      x
    </pre>
  `;
}

if (true) {
  if (true) {
    html`
      <pre>
        x
      </pre
      >
    `;
  }
}

```

Reply:
Any ideas?